sdk/confidentialledger/confidential-ledger-rest/test/public/colderEndpoints.spec.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  ConfidentialLedgerClient,
+  GetConsortiumMembers200Response,
+  GetEnclaveQuotes200Response,
+  isUnexpected,
+} from "../../src";
+import { createClient, createRecorder } from "./utils/recordedClient";
+
+import { Context } from "mocha";
+import { EnclaveQuoteOutput } from "../../src";
+import { Recorder } from "@azure-tools/test-recorder";
+import { assert } from "chai";
+
+describe("Colder endpoints", () => {
+  let recorder: Recorder;
+  let client: ConfidentialLedgerClient;
+
+  beforeEach(async function (this: Context) {
+    recorder = createRecorder(this);
+    client = await createClient();
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("should obtain constitution from ledger", async function () {
+    const result = await client.path("/app/governance/constitution").get();
+
+    assert.equal(result.status, "200");
+
+    if (isUnexpected(result)) {
+      // If this condition is true, TypeScript is able to infer that the type of result is
+      // GetConstitutiondefaultResponse because we got an "unexpected" response. If you hover over
+      // result, the right type is displayed and intellisense gives you the right options.
+      assert.fail(result.status, "200", `Unexpected response status ${result.status}`);
+    }
+
+    // Outside of the if statement, TypeScript infers that the response status was "200" and gives result a type of GetConstitution200Response
+    // without the need to cast
+    assert.typeOf(result.body.digest, "string");
+    assert.typeOf(result.body.script, "string");
+  });
+
+  it("should retrieve a list of consortium members", async function () {
+    const result = await client.path("/app/governance/members").get();
+
+    assert.equal(result.status, "200");
+
+    const memberResponse = result as GetConsortiumMembers200Response;
```

Comment:
What do you think about this change? I'd like to avoid casting types, but we can keep the explicit type annotation without casting, using the help of type narrowing and the `isUnexpected` helper. I'd suggest to do this in all the places where `as` is used
```suggestion
    if (isUnexpected()){
       throw result.body;
    }
    const memberResponse: GetConsortiumMembers200Response = result;
```